libmcount/mcount.c

```diff
@@ -1339,6 +1354,9 @@ static int __mcount_entry(unsigned long *parent_loc, unsigned long child, struct
 	rstack->nr_events = 0;
 	rstack->event_idx = ARGBUF_SIZE;
 
+	if (mcount_size_enabled)
+		rstack->size = mcount_getsize(&mcount_sym_info, child);
```

Comment:
Also we have `__cygprof_entry()` and `_xray_entry()`.  That why it's better to do it in `mcount_entry_filter_record()`.